processing/src/main/java/org/apache/druid/query/CacheStrategy.java

```diff
@@ -155,10 +157,16 @@ static void fetchAggregatorsFromCache(
         throw new ISE("Ran out of objects while reading aggregators from cache!");
       }
 
-      if (isResultLevelCache) {
-        addToResultFunction.apply(aggregator.getName(), i, resultIter.next());
-      } else {
+      ColumnType resultType = aggregator.getResultType();
+      ColumnType intermediateType = aggregator.getIntermediateType();
+
+      boolean needsDeserialize = !isResultLevelCache
+          || (!resultType.isPrimitive() && resultType.equals(intermediateType));
```

Comment:
if `resultType.isPrimitive() && resultType.equals(intermediateType)`  is *true*; then 
both `resultType.isPrimitive()` and `intermediateType.isPrimitive()` is true - so there might be not much to deserialize....

I've removed it - as I don't think it will make a big difference